### PR TITLE
chore(deps): update deps to latest, drop old beachball and eslint deps, add new eslint config file

### PR DESCRIPTION
**Pull request by ctran88**
## What's New?

- update deps to latest
  - except node-fetch, which causes some import issues in v3
- since eslint was updated from v8 to v9, had to update config file and rules
- moved the old e2e tests to be co-located next to their source files since eslint was complaining that it wasn't a part of the project service discovery
- dropped beachball

## Type of change

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] Documentation update

## Checklist:

- [ ] My code follows the style guidelines of this project
- [ ] I have performed a self-review of my own code
- [ ] I have manually tested my code thoroughly
- [ ] I have added/updated inline documentation for public facing interfaces if relevant
- [ ] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] New and existing integration and unit tests pass locally with my changes
- [ ] Any dependent changes have been merged and published in downstream modules

### CODE REVIEW

File: CHANGELOG.json
Comment (bertrmz) [inline CHANGELOG.json:1]: Does the data in this file need to be persisted in a different format?

File: src/classes/User/User.e2e.spec.ts [hunk 1]
```diff
@@ -0,0 +1,69 @@
...
+                id: userId,
+                createdAt: expect.any(Date),
+                lastLoginAt: expect.any(Date),
+                loginCount: expect.any(Number),
+                status: expect.any(String),
+                updatedAt: expect.any(Date),
+                userMetadata: expect.any(Object),
+                webauthn: expect.any(Boolean),
+                webauthnDevices: expect.any(Array<WebAuthnDevices>),
+                webauthnTypes: expect.any(Array<WebAuthnType>),
```
Comment (bertrmz) [inline src/classes/User/User.e2e.spec.ts:33]: The other matchers here seem to be using constructors of primitive JS types.
What's the expected behavior of an `expect.any(Array<FooType>)` match?

File: src/utils/apiConfiguration.ts [hunk 2]
```diff
@@ -13,7 +13,7 @@ export function apiConfiguration(config?: ConfigurationParameters): Configuratio
         fetchApi: fetch as unknown as ConfigurationParameters['fetchApi'],
         headers: {
             ...config?.headers,
-            'Authorization': `Bearer ${config?.accessToken}`,
+            'Authorization': `Bearer ${config?.accessToken as string}`,
```
Comment (bertrmz) [inline src/utils/apiConfiguration.ts:16]: Does the case of an undefined config need to be handled?

& a separate question, but why are `accessToken` and `headers.Authorization` both needed in this object?

File: CHANGELOG.json
Comment (ctran88) [reply CHANGELOG.json:1]: no the last published version already used the contents of this file, which is put into `CHANGELOG.md`. this is only a cache for beachball

File: src/classes/User/User.e2e.spec.ts (hunk 1 above)
Comment (ctran88) [reply src/classes/User/User.e2e.spec.ts:33]: it matches against an array of objects whose shape (keys and key value types) conform to `FooType`

File: src/utils/apiConfiguration.ts (hunk 2 above)
Comment (ctran88) [reply src/utils/apiConfiguration.ts]: `apiConfiguration` is only ever called internally when constructing codegen options https://github.com/passageidentity/passage-flex-node/blob/7dae6eb25411d8d3c021c70cda68dbbad15a5a35/src/classes/PassageFlex/PassageFlex.ts#L31

and we take in our own values in `PassageFlexConfig` to populate this function's parameters https://github.com/passageidentity/passage-flex-node/blob/ceabe0373a02bc25da8a151d45ed395deb8c568b/src/classes/PassageFlex/types.ts#L4

where we always throw on undefined for the api key before this function is called https://github.com/passageidentity/passage-flex-node/blob/7dae6eb25411d8d3c021c70cda68dbbad15a5a35/src/classes/PassageFlex/PassageFlex.ts#L23-L24

so the value should never be undefined at this point. i changed the parameter to be required in 6c9991634979910148cdf62528094e81492f91b5

Comment (ctran88) [reply src/utils/apiConfiguration.ts]: > why are accessToken and headers.Authorization both needed in this object?

good point, i've removed the header in de63778b98752e5165925c597932c2356db324a0

**Review by bertrmz — APPROVED** [review]

**Pull request merged**